enhancements/sig-architecture/167-token-based-registration/README.md

```diff
@@ -0,0 +1,391 @@
+
+1. **Simplified Setup**: Token-based auth requires less infrastructure than CSR signing or cloud-specific IAM integration
+2. **Kubernetes-Native**: Leverages built-in Kubernetes TokenRequest API without external dependencies
+3. **Automatic Rotation**: Tokens can be automatically refreshed without manual intervention
+4. **Lower Overhead**: No need for certificate management, CSR approval workflows, or cloud provider integration
+5. **Development and Testing**: Easier to set up for local development and testing scenarios
+
+### Goals
+
+- Implement a token-based registration driver using Kubernetes TokenRequest API
+- Support managed cluster registration and add-on registration independently
+  - Cluster and add-on registration can use different drivers (e.g., cluster uses CSR, add-ons use token)
+  - Provides flexibility to balance security and operational requirements for each component
+- Support automatic token refresh before expiration
+- Maintain security through scoped RBAC permissions and expiration enforcement
+- Follow existing driver patterns for consistency
+- Enable simpler bootstrap process for managed clusters and add-ons
+
+### Non-Goals
+
+- Replace existing CSR, AWS IRSA, or GRPC drivers (this is an additional option)
+- Support external token providers (OIDC, OAuth) in initial implementation
+- Implement token revocation mechanisms (future enhancement)
+- Support short-lived tokens (< 1 hour) in initial implementation
+
+## Proposal
+
+### User Stories
+
+#### Story 1: Simplified Local Development
+As a developer working on OCM features, I want to quickly register local test clusters without setting up CSR approval or cloud IAM infrastructure, so I can iterate faster on development.
+
+#### Story 2: Clusters with Difficult CSR Approval
+As a platform operator managing clusters where CSR approval is difficult or restricted (such as Amazon EKS), I want an alternative authentication mechanism that doesn't require managing certificate signing infrastructure, so I can register and manage clusters without operational friction.
+
+#### Story 3: SSL Termination at Proxy
+As a platform operator with security requirements where connections from managed clusters to the hub go through a load balancer, API gateway, or reverse proxy that terminates SSL connections, I need an authentication mechanism that works without client certificate authentication, since the SSL termination breaks certificate-based authentication (CSR).
+
+### Implementation Details
+
+This section describes the implementation details for both cluster registration and add-on registration using the token driver.
+
+#### Cluster Registration
+
+##### User Configuration
+
+Enable token-based registration in the Klusterlet resource:
+
+```yaml
+apiVersion: operator.open-cluster-management.io/v1
+kind: Klusterlet
+metadata:
+  name: klusterlet
+spec:
+  clusterName: cluster1
+  registrationConfiguration:
+    registrationDriver:
+      authType: token
```

Comment:
I think we should have a separated addonDriver

Reply:
Removed the token based cluster registration from the proposal.